Makefile

```diff
@@ -13,11 +15,18 @@ luacheck:
 	luacheck --config .luacheckrc .
 
 test_integration:
-	$(PYTHON) test/test-run.py --force $(TEST_RUN_EXTRA_PARAMS)
+	PYTHONPATH=$(PROJECT_DIR) $(PYTHON) test/test-run.py --force $(TEST_RUN_EXTRA_PARAMS)
 
 test_unittest:
 	$(PYTHON) -m unittest discover test/unittest/
 
 test: test_unittest test_integration
 
+coverage:
+	PYTHON="$(PYTHON) -m coverage run" make -f $(MAKEFILE_PATH) test
```

Comment:
This is not clear for me, to be honest. Can't we run it as `coverage run make -f $(MAKEFILE_PATH) test` (the documentation suggests this variant)? If so, is there a difference with given variant that uses `PYTHON="$(PYTHON) -m coverage run"`?

Reply:
> Can't we run it as `coverage run make -f $(MAKEFILE_PATH) test` (the documentation suggests this variant)? 
when you want to enable profiling and gather code coverage you should use `coverage run` instead of `python`,
you don't use python to execute make with certain Makefile, but if you will try you will get an error:
```
sergeyb@pony:~/sources/MRG/tarantool/test-run$ python make -f Makefile 
python: can't open file 'make': [Errno 2] No such file or directory
sergeyb@pony:~/sources/MRG/tarantool/test-run$ 
```

>  If so, is there a difference with given variant that uses PYTHON="$(PYTHON) -m coverage run"?
AFAIR without $PYTHON it doesn't work. Let's try to remove $PYTHON:

```
--- a/Makefile
+++ b/Makefile
@@ -23,7 +23,7 @@ test_unittest:
 test: test_unittest test_integration
 
 coverage:
-       PYTHON="$(PYTHON) -m coverage run" make -f $(MAKEFILE_PATH) test
+       PYTHON="coverage run" make -f $(MAKEFILE_PATH) test
        coverage report
 
 clean:
sergeyb@pony:~/sources/MRG/tarantool/test-run$ make coverage 
PYTHON="coverage run" make -f /home/sergeyb/sources/MRG/tarantool/test-run/Makefile test
make[1]: Entering directory '/home/sergeyb/sources/MRG/tarantool/test-run'
coverage run -m unittest discover test/unittest/
E..
======================================================================
ERROR: test_extract_schema_from_snapshot (test_lib_tarantool_connection.TestTarantoolConnection)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/sergeyb/sources/MRG/tarantool/test-run/test/unittest/test_lib_tarantool_connection.py", line 8, in test_extract_schema_from_snapshot
    v = utils.extract_schema_from_snapshot(snapshot_path)
NameError: global name 'utils' is not defined

----------------------------------------------------------------------
Ran 3 tests in 0.036s

FAILED (errors=1)
```

Reply:
I was wrong, without $PYTHON it works fine:
```
--- a/Makefile
+++ b/Makefile
@@ -23,7 +23,7 @@ test_unittest:
 test: test_unittest test_integration
 
 coverage:
-       PYTHON="$(PYTHON) -m coverage run" make -f $(MAKEFILE_PATH) test
+       PYTHON="coverage run" make -f $(MAKEFILE_PATH) test
        coverage report
 
 clean:
```

Reply:
My question was why not just `coverage run make test`, but now I see: `coverage run` expects a python script as argument and does not accept a wrapper around python script. Okay so.